### PR TITLE
Add toMap and fromMap to Data.Set.

### LINKED ISSUE #20: Efficient conversion functions between Set and Map
Since the implementation of `Set` is based on `Map` it would be nice to expose conversions functions. In this case these are only the constructor and projection of the newtype.

### PR DESCRIPTION
**Pull request by np**
Fixes #20.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -26,8 +26,8 @@
     "purescript-st": "^4.0.0",
     "purescript-tailrec": "^4.0.0",
     "purescript-tuples": "^5.0.0",
+    "purescript-typelevel-prelude": "^5.0.1",
     "purescript-unfoldable": "^4.0.0",
```
Comment (kritzcreek) [inline bower.json]: the trailing comma here is killing CI

**Comment by np** [comment]
I'm not sure how to fix the remaining issues.

**Comment by np** [comment]
Can we turn warn-error only on the code of this package? I don't see the point of failing the build because of another package warnings.

**Comment by hdgarrood** [comment]
There's also a real error in the CI logs, which will need fixing:

```
bower ECONFLICT     Unable to find suitable version for purescript-typelevel-prelude
```

The `--censor-lib` flag is already set in the `build` script in `package.json`, so warnings from dependencies shouldn't cause a build to fail. For some reason, that particular file is being treated as part of this package by `psa`. I'm not sure why that is.

**Comment by kl0tl** [comment]
Done in https://github.com/purescript/purescript-ordered-collections/pull/31.

**Pull request closed without merging**